src/types/parse_mode.rs

```diff
@@ -74,6 +76,45 @@ pub enum ParseMode {
     Markdown,
 }
 
+impl TryFrom<&str> for ParseMode {
+    type Error = ();
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let normalized = value.to_lowercase();
+        match normalized.as_ref() {
+            "html" => Ok(ParseMode::HTML),
+            "markdown" => Ok(ParseMode::Markdown),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<String> for ParseMode {
+    type Error = ();
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
```

Comment:
You can just delegate to `TryFrom<&str>` like `Self::try_from(&value)`. I'm actually not sure if we even need `TryFrom<String>` impl, if we have `TryFrom<&str>`